docs/html/reference/build-system/setup-py.md

```diff
@@ -72,7 +72,7 @@ editable/development installation.
 
 ### Cleanup
 
-After attempting installation, pip may run `setup.py clean` to clean up build
+After generating a wheel, pip may run `setup.py clean` to clean up build
```

Comment:
After failing to build a wheel ?

Reply:
https://github.com/pypa/pip/blob/e1dd4b425505fa690fb45d84d1434fc2e8e13eac/src/pip/_internal/wheel_builder.py#L301-L303

It's done outside of the try-except, unconditionally as far as I can tell.

Reply:
But if successful there is a return at line 294 so the cleanup part is not reached ?